### PR TITLE
mgr/MgrClient: fix reconnect event leak

### PR DESCRIPTION
**Pull request by liewegas**
http://tracker.ceph.com/issues/19580

### CODE REVIEW

File: src/mgr/MgrClient.cc [hunk 1]
```diff
@@ -49,6 +49,11 @@ void MgrClient::shutdown()
 {
   Mutex::Locker l(lock);
 
+  if (connect_retry_callback) {
```
Comment (tchaikov) [inline src/mgr/MgrClient.cc]: how about `report_callback`?

**Comment by cbodley** [comment]
oh cool, i just saw this leak in the rgw suite. i opened http://tracker.ceph.com/issues/19580 for it

**Comment by liewegas** [comment]
@tchaikov added report_callback cleanup
@cbodley added link to ticket in commit

File: src/mgr/MgrClient.cc [hunk 2]
```diff
@@ -49,6 +49,15 @@ void MgrClient::shutdown()
...
+    timer.cancel_cevent(connect_retry_callback);
```
Comment (jcsp) [inline src/mgr/MgrClient.cc]: s/cancel_cevent/cancel_event/

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-04-17_21:11:42-rados-wip-sage-testing2_2017_4_18-distro-basic-smithi

http://pulpito.ceph.com/yuriw-2017-04-18_14:45:29-rados-wip-sage-testing2_2017_4_18---basic-smithi/

**Pull request merged**